ubi_manifest/app/utils.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from typing import Any
+
+import ubiconfig
+from pubtools.pulplib import Client, Criteria
+
+from ubi_manifest.worker.tasks.depsolver.utils import make_pulp_client
+
+_LOG = logging.getLogger(__name__)
+
+
+def get_items_for_depsolving(
+    app_conf: Any, provided_repos: list[str]
+) -> list[dict[str, Any]]:
+    """
+    Returns a list of all {"repo_group": ["repo1, repo2], "url" = "https://config"}
+    items which are then used for creation of depsolving tasks.
+    """
+    all_items: list[dict[str, Any]] = []
+
+    # client-tools repos and ubi repos need to be handled separately
+    client_tools_repos = [repo for repo in provided_repos if "client-tools" in repo]
```

Comment:
This looks a bit problematic, I can imagine we can get request to make different repos public (`xyz-public-tools` ) and use ubi-manifest for calculating content. So this will stop working and we will need to do extra fix for each addition of new class of repos.

Do you think this can be re-written in a more generic way. Currently we understand that there are two strategies available for creating manifest. Strategy is driven by _base_pkgs_only_ flag in input yaml files.
If this is set to true, we don't do any depsolving - we'll just calculate content based on pkgs defined in yaml.

If the flag is not present or set to True, ubi-manifest will do full depsolving where we typically need to pull  dependencies from other repos which was previously defined by `allowed_repo_groups` config. 

So can you please re-write this in a way so:

1. we don't use specific keywords (ubi, client_tools) and
2. the determination of what class of repositories we are processing is driven by _base_pkgs_only_ flag ?